src/main/java/com/sprint/mission/discodeit/entity/ReadStatus.java

```diff
@@ -12,32 +12,38 @@
 @Getter
 @Entity
 @Table(
-        name = "read_statuses",
-        uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "channel_id"})
+    name = "read_statuses",
+    uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "channel_id"})
 )
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
 public class ReadStatus extends BaseUpdatableEntity implements Serializable {
 
-    @ManyToOne(fetch = FetchType.LAZY,optional = false)
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
     @JoinColumn(name = "user_id", columnDefinition = "UUID")
     private User user;
 
-    @ManyToOne(fetch = FetchType.LAZY,optional = false)
-    @JoinColumn(name = "channel_id",columnDefinition = "UUID")
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "channel_id", columnDefinition = "UUID")
     private Channel channel;
 
-    @Column(columnDefinition = "TIMESTAMP WITH TIME ZONE",nullable = false)
+    @Column(columnDefinition = "TIMESTAMP WITH TIME ZONE", nullable = false)
     private Instant lastReadAt;
 
+    private boolean notificationEnabled;
+
     public ReadStatus(User user, Channel channel, Instant lastReadAt) {
         this.user = user;
         this.channel = channel;
         this.lastReadAt = lastReadAt;
+        this.notificationEnabled = channel.getType() == ChannelType.PRIVATE;
```

Comment:
@coderimspace 생성자에서 비즈니스 로직을 처리하게 되면 테스트하기가 어려워집니다. 팩토리 메서드 패턴을 써보면 어떨까요?
```suggestion
        public static ReadStatus createWithDefaultNotification(User user, Channel channel, Instant lastReadAt) {
    ReadStatus readStatus = new ReadStatus(user, channel, lastReadAt);
    readStatus.notificationEnabled = channel.getType() == ChannelType.PRIVATE;
    return readStatus;
}
```